backtrace-library/src/main/java/backtraceio/library/breadcrumbs/BacktraceBreadcrumbs.java

```diff
@@ -0,0 +1,342 @@
+            backtraceReport.attachmentPaths.add(BacktraceBreadcrumbs.getBreadcrumbLogPath(context));
+
+            long lastBreadcrumbId = this.getCurrentBreadcrumbId();
+            backtraceReport.attributes.put("breadcrumbs.lastId", lastBreadcrumbId);
+        }
+    }
+
+    /**
+     * Create a breadcrumb which reflects the current breadcrumb configuration
+     * @return true if the breadcrumb was successfully added
+     */
+    private boolean addConfigurationBreadcrumb()
+    {
+        if (backtraceBreadcrumbsLogger == null) {
+            return false;
+        }
+
+        Map<String, Object> attributes = new HashMap<String, Object>();
+
+        for (BacktraceBreadcrumbType enabledType : BacktraceBreadcrumbType.values())
+        {
+            if (enabledType == BacktraceBreadcrumbType.CONFIGURATION) {
+                attributes.put(enabledType.toString(), "enabled");
+            }
+
+            String state = (enabledBreadcrumbTypes != null &&
+                    enabledBreadcrumbTypes.contains(enabledType)) ? "enabled" : "disabled";
+
+            attributes.put(enabledType.toString(), state);
+        }
+
+        return backtraceBreadcrumbsLogger.addBreadcrumb("Breadcrumbs configuration",
+                attributes,
+                BacktraceBreadcrumbType.CONFIGURATION,
+                BacktraceBreadcrumbLevel.INFO);
+    }
+
+    private void handleExistingBreadcrumbLogFile(Context context, BacktraceBase breadcrumbLogSender) {
+        if (context == null || breadcrumbLogSender == null) {
+            return;
+        }
+
+        final File breadcrumbLogFile = new File(BacktraceBreadcrumbs.getBreadcrumbLogPath(context));
+        final File breadcrumbTempLogFile = new File(this.getBreadcrumbLogDirectory() + "/bt-stalecrumbs");
+        try {
+            if (breadcrumbLogFile.exists()) {
+                FileUtils.moveFile(breadcrumbLogFile, breadcrumbTempLogFile);
+                BacktraceReport staleBreadcrumbsReport = new BacktraceReport("Found breadcrumbs from a previous session");
+                staleBreadcrumbsReport.attachmentPaths.add(breadcrumbTempLogFile.getAbsolutePath());
+
+                OnServerResponseEventListener staleBreadcrumbSubmissionCompleteCallback = new OnServerResponseEventListener() {
+                    @Override
+                    public void onEvent(BacktraceResult backtraceResult) {
+                        breadcrumbTempLogFile.delete();
+                    }
+                };
+
+                breadcrumbLogSender.send(staleBreadcrumbsReport, staleBreadcrumbSubmissionCompleteCallback);
+            }
+        } catch (Exception ex) {
+            BacktraceLogger.e(LOG_TAG, "Could not send previously existing breadcrumbs due to: " + ex.getMessage());
+            breadcrumbLogFile.delete();
+        }
+    }
+
+    private boolean isBreadcrumbsEnabled()
+    {
+        return enabledBreadcrumbTypes != null && !enabledBreadcrumbTypes.isEmpty();
```

Comment:
if we use breadcrumb flag instead of set of possible enums, this condition will be much easier to understand - or actually we won't need to use this method at all - breadcrumbType != breadcrumbType.None